### PR TITLE
Bug fix for vnic backing device remove failure and vnic auto failover

### PR DESCRIPTION
**Pull request by vaishnavibhat**
Fixing below issues:
- vnic backing device remove failure is seen when auto failover test is run for odd timings. Addressing this issue by setting back the priority
- Setting back original priorities after vnic auto failover which sets the priority to 1 and 100
- Seeting back active interface in teardown after every tests

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
avocado run --test-runner runner network_virtualization.py -m network_virtualization.yaml
JOB ID     : 1066ae8569efb24b1997170d27237cd1f0aa96f2
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-10-10T11.03-1066ae8/job.log
 (01/11) network_virtualization.py:NetworkVirtualization.test_add;run-f043: PASS (19.26 s)
 (02/11) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-f043: PASS (16.24 s)
 (03/11) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-f043: FAIL: test failed,check dmesg log in debug log (18.80 s)
 (04/11) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-f043: PASS (78.59 s)
 (05/11) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-f043: PASS (136.19 s)
 (06/11) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-f043: PASS (85.77 s)
 (07/11) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-f043: PASS (87.28 s)
 (08/11) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-f043: PASS (58.44 s)
 (09/11) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-f043: PASS (63.28 s)
 (10/11) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-f043: PASS (11.28 s)
 (11/11) network_virtualization.py:NetworkVirtualization.test_remove;run-f043: PASS (16.38 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-10-10T11.03-1066ae8/results.html
JOB TIME   : 599.16 s

[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9747544/job.log)

File: io/net/virt-net/network_virtualization.py [hunk 1]
```diff
@@ -928,6 +953,17 @@ def get_active_device_logport(self, slot):
                         break
         return logport
 
+    def get_backing_device_priority(self, slot):
+        '''
+        Get the backing device proiority of the vnic interface
+        '''
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:959]: docstring, same here

File: io/net/virt-net/network_virtualization.py [hunk 2]
```diff
@@ -928,6 +961,17 @@ def get_active_device_logport(self, slot):
...
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:967]: @vaishnavibhat you missed the docstrig part here.

File: io/net/virt-net/network_virtualization.py [hunk 3]
```diff
@@ -917,6 +946,10 @@ def get_active_device_logport(self, slot):
         '''
         Get the logical port id of the Network
         virtualized device
+        :param slot: vnic slot
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:949]: please maintain the proper line spaces here after desciption

File: io/net/virt-net/network_virtualization.py [hunk 4]
```diff
@@ -350,6 +354,20 @@ def is_disabled_enabled_dev(self, expect):
                 else:
                     self.fail("Could not %s vNIC interface" % operation)
 
+    def change_active_device(self, slot):
+        '''
+        Change active device to original active device
+        :param slot: vnic slot
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:360]: proper line spaces after description

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Review by Naresh-ibm — APPROVED** [review]
@vaishnavibhat thank you for adding the changes suggest.
LGTM

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@vaishnavibhat cant calling update_backing_device() function inside est_backingdevremove(self): ? will not solve this problem.. https://github.com/avocado-framework-tests/avocado-misc-tests/blob/af4918570d998f7d7db87e704edf334251d68f5a/io/net/virt-net/network_virtualization.py#L777

instead of new function

**Comment by vaishnavibhat** [comment]
> @vaishnavibhat cant calling update_backing_device() function inside test_backingdevremove(self): ? will not solve this problem..
> 
> https://github.com/avocado-framework-tests/avocado-misc-tests/blob/af4918570d998f7d7db87e704edf334251d68f5a/io/net/virt-net/network_virtualization.py#L777
> 
> instead of new function

update_backing_devices() will update the information of the active device only. The function has does not record the latest priority of the active and backing device. 
  Also , if some other tests changes the priority to a different value apart from the ones mentioned by user in yaml file, this function fails to set it to original values. Hence the backing device and original device are not set like to original devices.

**Comment by vaishnavibhat** [comment]
Resent the PR with requested changes. 

In test 1 (test_remove) and test 11 (test_remove), The tests will PASS but the logs will report ' Unable to set back the original active device' in the debug.log since there is only single backing device. 
Rest of the tests will either report 'Unable to set back the original active device' if failed to set back, or a success if the backing device is set back to original. 

Thank you.

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**